ts/resolution.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2014 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Matt Tracy (matt.r.tracy@gmail.com)
+
+package ts
+
+import (
+	"fmt"
+	"time"
+)
+
+// Resolution is used to enumerate the different resolution values supported by
+// Cockroach.
+type Resolution int64
+
+// Resolution enumeration values are directly serialized and persisted into
+// system keys; these values must never be altered or reordered.
+const (
+	// Resolution10s stores data with a sample resolution of 10 seconds.
+	Resolution10s Resolution = 1
+)
+
+// sampleDurationByResolution is a map used to retrieve the sample duration
+// corresponding to a Resolution value. Sample durations are expressed in
+// nanoseconds.
+var sampleDurationByResolution = map[Resolution]int64{
```

Comment:
The value of `Resolution` is encoded into the key, so this kind of sucks. If we're worried about performance here (which we shouldn't be yet), all of these maps could be slices:

```
var sampleDurationByResolution = [...]int64{
  Resolution10s: int64(time.Second * 10)
}
```